### DISCUSSION TITLE
Help with regex on APK version

### DISCUSSION THREAD

**Discussion by Taras-Ivashchuk** [ImranR98/Obtainium#1887] [Q&A] [answered]
Got tutonata app installed. Tutanota [repo](https://github.com/tutao/tutanota/releases) started to host apps for calendar builds, but I don't care about it, since I need only a sheer tutanota app for Android. I've been budging around for a long time, but still end up with calendar app.  I had tried to edit APK version to have a lookaround regex  like ^((?!.*webview.*).)*$ , but it didn't help me a bit. I either edited release's title and release notes, however got the same result. The simple regex tutanota.*.apk still fetches calendar,  and I  don't know why though , feeling jaded to entertain something.

**Comment by DarkCrypt** [comment, marked as answer]
I see what's going on here. There's a few options.

**Option (1)**
Your webview ReGex is wrong and it won't work for this particular repo anyway. If your ReGex was correct it would filter out release titles for webview not calendar.

To make it easy to understand, here's a basic ReGex to eliminate a release title within GitHub: `^((?!.*whateverword.*).)*$`

All we need to do is change `whateverword` to `Calendar` so it's like this...

Filter Release Titles by Regular Expression: `^((?!.*Calendar.*).)*$`

*Capitalization matters. A lower case "c" won't work, it must match the release title within GitHub.

**Option (2)**
It's possible to just use a simple filter as you would for other repos like this...

Filter APKs by Regular Expression: `tutanota-app`

**Comment by Taras-Ivashchuk** [comment]
The ReGex ^((?!.*Calendar.*).)*$ works eliminating the Calendar title , but it fetches a next release title, which is for iOS, so after adjusting it to ^((?!.*(Calendar|iOS).*).)*$ got it at last to what I strived for.

**Reply by DarkCrypt** [reply]
I tried both of your ReGex and got errors.

**Reply by Taras-Ivashchuk** [reply]
Here I used `^((?!.(Calendar|iOS).).)$` in a release title field.  And it turned out to be working, however,  have to manually click to check the apk version, not the archive.
![Screenshot_20241007-121951](https://github.com/user-attachments/assets/12a04b43-74c0-4d98-9767-4e8ecbd9a16c)
![Screenshot_20241006-154254](https://github.com/user-attachments/assets/8306a0c0-e77d-47a1-ab7b-b8e035baed29)

**Reply by DarkCrypt** [reply]
Every ReGex you type throws errors but the ReGex in the screenshot with the added `*` characters seem to work. Add option 2 from my above comment and that should eliminate the selector.
